geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureConverterFactory.scala

```diff
@@ -139,16 +140,21 @@ trait ToSimpleFeatureConverter[I] extends SimpleFeatureConverter[I] with Logging
 
   val reuse = Array.ofDim[Any](requiredFields.length)
 
-  def processSingleInput(i: I): Option[SimpleFeature] =
+  def processSingleInput(i: I, gParams: Option[Map[String, String]] = None): Option[SimpleFeature] = {
+    if (gParams.isDefined) ctx.globalParams = gParams
     Try { convert(fromInputType(i), reuse) } match {
       case Success(s) => Some(s)
       case Failure(t) =>
         logger.debug("Failed to parse input", t)
         None
     }
+  }
 
-  def processInput(is: Iterator[I]): Iterator[SimpleFeature] =
+  def processInput(is: Iterator[I], gParams: Option[Map[String, String]] = None): Iterator[SimpleFeature] = {
+    if (gParams.isDefined) ctx.globalParams = gParams
```

Comment:
This is where you should call resetCount.  Also, see above re: passing in an empty map.